### DISCUSSION TITLE
every possible space group setting

### DISCUSSION THREAD

**Discussion by aslarsen** [spglib/spglib#273] [Q&A] [answered]
Hi
If I have a crystal structure in space group P21/c (14) is there a way of generating the coordinates and lattices of the crystal structure in every possible space group setting in P21/c? using spglib.

**Comment by atztogo** [comment]
I think you can use `spg_get_dataset_with_hall_number` where you may specify id numbers of hall symbol from 81 to 89. The number is found e.g. at https://yseto.net/?page_id=29&lang=en.

**Reply by aslarsen** [reply]
Thanks, this seems not to be available with the python interface? And is there any easy way to add it?

**Reply by atztogo** [reply, marked as answer]
In the case of python, it is `get-symmetry-dataset` with `hall_number` as an optional parameter,

https://spglib.readthedocs.io/en/latest/python-spglib.html#get-symmetry-dataset

Be careful that if the `hall_number` doesn't correspond to the space group type, spglib gets stuck if I remember correctly. I have to fix this in the future.

**Comment by husakm** [comment]
Generally spglib does not cover all possible space groups settings. There are used much more SP setting than the 530 Hall one. 
Eg. for P1  B1, F1 can be used ... For P-1 B-1,C-1,F-1,I-1 can be used ... Etc. It does not give sense, but it is used in reality (with access to CSD you can find structures published with such settings) ...

**Reply by atztogo** [reply]
I don't know the details about what you write. But do you mean you want to contribute implementing those in spglib?

**Comment by husakm** [comment]
Sorry for previous answer related to another topic ....
Yes, we work on a spg database covering all non standard settings included in Cambridge database ....
After finalizing + testing, we can discuse how to contrubute this to spglib ...
